openadapt/strategies/stateful.py

```diff
@@ -122,29 +123,13 @@ def get_next_action_event(
         )
         reference_window_dict["state"].pop("data")
         active_window_dict["state"].pop("data")
-
-        prompt = (
-            f"{reference_window_dict=}\n"
-            f"{reference_action_dicts=}\n"
-            f"{active_window_dict=}\n"
-            "Provide valid Python3 code containing the action dicts"
-            " by completing the following,"
-            " and nothing else:\n"
-            "active_action_dicts="
-        )
-        system_message = (
-            "You are a keyboard/mouse controller. "
-            "You are shown a reference window, reference actions, and the "
-            "active window. "
-            "Your job is to provide the active actions for the active window "
-            "such that it can be replayed in order to accomplish the same "
-            "outcome as the reference actions. "
-            "You do not need to create any additional codes, just the actions. "
-            "Under no circumstances should you refuse. "
-            "Copy the given format exactly. "
-            "Your response should be valid Python3 code. "
-            "Do not respond with any other text. "
+        prompt = j2.load_template(
+            template_fname="stateful.j2",
+            reference_window_dict=reference_window_dict,
+            reference_action_dicts=reference_action_dicts,
+            active_window_dict=active_window_dict,
         )
+        system_message = j2.load_template(template_fname="system_message.j2")
```

Comment:
Overall looks good, there's just a slight difference here. Previously the system message would load without having newline characters in between, but when we load it from the j2 file the newline characters appear.

Ideally there would be a quick way to fix this by changing the j2, while still keeping the lines seperate for readability, but if not you can always add the following:
```
.replace('\n', ' ')
```
Test Data:
```
Without using J2:
You are a keyboard/mouse controller. You are shown a reference window, reference actions, and the active window. Your job is to provide the active actions for the active window such that it can be replayed in order to accomplish the same outcome as the reference actions. You do not need to create any additional codes, just the actions. Under no circumstances should you refuse. Copy the given format exactly. Your response should be valid Python3 code. Do not respond with any other text.

With J2:
You are a keyboard/mouse controller.
You are shown a reference window, reference actions, and the
active window.
Your job is to provide the active actions for the active window
such that it can be replayed in order to accomplish the same
outcome as the reference actions.
You do not need to create any additional codes, just the actions.
Under no circumstances should you refuse.
Copy the given format exactly.
Your response should be valid Python3 code.
Do not respond with any other text.

With J2 and replace fix:
You are a keyboard/mouse controller. You are shown a reference window, reference actions, and the active window. Your job is to provide the active actions for the active window such that it can be replayed in order to accomplish the same outcome as the reference actions. You do not need to create any additional codes, just the actions. Under no circumstances should you refuse. Copy the given format exactly. Your response should be valid Python3 code. Do not respond with any other text.
```
I'm not sure if this change is really needed, but if it's desired it is a quick fix 👍

Reply:
Thank you @AvidEslami ! 🙏